sycl/plugins/hip/pi_hip.cpp

```diff
@@ -3912,7 +3911,8 @@ pi_result hip_piEnqueueMemBufferFill(pi_queue command_queue, pi_mem buffer,
         value = *(static_cast<const uint8_t *>(pattern) + step);
 
         // offset the pointer to the part of the buffer we want to write to
-        auto offset_ptr = dstDevice + (step * sizeof(uint8_t));
+        auto offset_ptr = reinterpret_cast<void *>(
```

Comment:
Looks like NFC change. How does it make possible to relax >4 byte patterns?

Reply:
Yes, this does not change any of the behaviour, you could use >4 bytes before for AMD. However this was failing when building HIP for an nvidia platform. This was caused due to `hipDeviceptr_t` being an `unsigned int` for NVIDIA or `void *`for AMD devices. For offsetting the pointer it should be a `unsigned int` and to call the function `hipMemset2DAsync`it needs to be `void*`. This is the opposite for 1D memsets which ask for `hipDeviceptr_t`.
So, to have the program handle both underlying device pointer types and different ptr arguments in each function you need to do some conversion that is not actually doing any work.


`hipError_t hipMemset2DAsync(void* dst, size_t pitch, int value, size_t width, size_t height,hipStream_t stream __dparm(0))`
`hipError_t hipMemsetD32Async(hipDeviceptr_t dst, int value, size_t count, hipStream_t stream)`